src/main/scala/firrtl/backends/verilog/VerilogEmitter.scala

```diff
@@ -1022,71 +1022,105 @@ class VerilogEmitter extends SeqTransform with Emitter {
           }
           instdeclares += Seq(");")
         case sx: DefMemory =>
+          require(sx.writeLatency == 1)
+          require(sx.readLatency <= 1)
           val options = emissionOptions.getMemoryEmissionOption(moduleTarget.ref(sx.name))
           val fullSize = sx.depth * (sx.dataType match {
             case GroundType(IntWidth(width)) => width
           })
           val decl = if (fullSize > (1 << 29)) "reg /* sparse */" else "reg"
           declareVectorType(decl, sx.name, sx.dataType, sx.depth, sx.info)
           initialize_mem(sx, options)
-          if (sx.readLatency != 0 || sx.writeLatency != 1)
+          if (sx.readLatency > 1 || sx.writeLatency != 1)
             throw EmitterException(
-              "All memories should be transformed into " +
+              "All complex memories should be transformed into " +
                 "blackboxes or combinational by previous passses"
```

Comment:
Could we add more helpful info here?
Like maybe mention `--repl-seq-mem` and `VerilogMemDelay`

Reply:
Done

Reply:
Done